src/main/java/com/diffmin/patch/PatchGeneration.java

```diff
@@ -162,10 +162,18 @@ public int getSrcNodeIndex(CtElement srcNode) {
                     }
                 };
         int srcNodeIndex = iUtil.getSrcNodeIndex(insertedNode);
-        CtElement parentElementInPrevModel = mapping.get(insertedNode.getParent());
+        CtElement parentElementInPrevModel = getMappingOfParent(insertedNode, mapping);
         return new ImmutableTriple<>(srcNodeIndex, insertedNode, parentElementInPrevModel);
     }
 
+    private static CtElement getMappingOfParent(CtElement insertedNode, SpoonMapping mapping) {
+        CtElement parentElementInPrevModel = mapping.get(insertedNode.getParent());
+        if (parentElementInPrevModel == null) {
+            return insertedNode.getParent();
+        }
+        return parentElementInPrevModel;
+    }
+
```

Comment:
There are cases when we move a node into _an inserted node_. This function handles that case. See https://github.com/SpoonLabs/diffmin/issues/95.